### PR TITLE
Issue 83 Store the best score in firebase immediately when the user finishes the game

### PR DESCRIPTION
**Pull request by alevarn**
Solves issue #43 #83 #90 #93 

- The leaderboard should now show the updated scores after a game has been completed.
- When the firebaseModel is updated with a new `playerHistory` it will update the Model, which will ensure that the `/profile` page has the latest scores.
- I also solved some small issues.

### CODE REVIEW

File: music-quiz/src/presenters/gamePresenter.js [hunk 1]
```diff
@@ -181,7 +182,7 @@ function GamePresenter(props) {
           <GameChallengeView
             playlist={challengePlaylistPromiseState.data}
             onStartClick={startGame}
-            onDeclineClick={declineChallenge}
+            onDeclineClick={clearGameSettings}
```
Comment (AnnaSkantz) [inline music-quiz/src/presenters/gamePresenter.js:185]: Do we not need to remove the request if it is declined? Like we did in "declineChallenge" (removed now).

Comment (AnnaSkantz) [reply music-quiz/src/presenters/gamePresenter.js:185]: Tested if I was able to decline a challenge and checked that the pending request was removed. This works so I guess this is not an issue and that you solve this another way 👍

**Review by AnnaSkantz — APPROVED** [review]
LGTM! Tested the functionality and it seems to work as expected. 👍

**Review by kanova98 — APPROVED** [review]
LGTM

**Pull request merged**